src/react/components/pages/editorPage/editorPage.tsx

```diff
@@ -142,9 +187,17 @@ export default class EditorPage extends React.Component<IEditorPageProps, IEdito
         if (!this.state.project && this.props.project) {
             this.setState({
                 project: this.props.project,
-                additionalSettings: { videoSettings: (this.props.project) ? this.props.project.videoSettings : null },
+                additionalSettings: { videoSettings: (this.props.project) ? this.props.project.videoSettings : null,
+                    activeLearningSettings: (this.props.project) ? this.props.project.activeLearningSettings : null },
             });
         }
+
+        if (this.state.project &&
+            this.state.project.activeLearningSettings.autoDetect &&
+            this.state.selectedAsset &&
```

Comment:
Consider encapsulating this into a variabled named `shouldPredictModel` or similar to make it easier to understand what is happening here.

We also should not be running predictions for every state change.  We should only predict when the selected asset has changed.

Reply:
@PIC123 implemented this but I see we are not running prediction on every state change.  We are checking at every state change if automatic prediction is enabled and if the asset eventually has been already predicted once before.  Is there another event handler where to put this logic other then componentDidUpdate() ?